src/stdatamodels/model_base.py

```diff
@@ -228,10 +228,6 @@ def __init__(
             asdffile = fits_support.from_fits(init, self._schema, self._ctx, **kwargs)
 
         elif isinstance(init, (str, bytes, PurePath)):
-            if isinstance(init, PurePath):
-                init = str(init)
-            if isinstance(init, bytes):
-                init = init.decode(sys.getfilesystemencoding())
```

Comment:
I think removing this results in passing a `bytes` instance to `asdf.open` (on line 244). This isn't supported. This might be "scope creep" but why are we supporting `bytes` here?